utils/src/test/java/pm/gnosis/utils/NumberUtilsKtTest.kt

```diff
@@ -0,0 +1,150 @@
+package pm.gnosis.utils
+
+import org.junit.Test
+
+import org.junit.Assert.*
+import pm.gnosis.tests.utils.Asserts.assertThrow
+import pm.gnosis.utils.exceptions.InvalidAddressException
+import java.math.BigDecimal
+import java.math.BigInteger
+
+class NumberUtilsKtTest {
+
+    @Test
+    fun hexAsEthereumAddressOrNull() {
+        assertNull("ffffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddressOrNull())
+        assertNull("0xffffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddressOrNull())
+        assertNull("thisisporbablynotahexnumber".hexAsEthereumAddressOrNull())
+        assertEquals(BigInteger("ffffffffffffffffffffffffffffffffffffffff", 16), "ffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddressOrNull())
+        assertEquals(BigInteger("ffffffffffffffffffffffffffffffffffffffff", 16), "0xffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddressOrNull())
+        assertEquals(BigInteger("abcd", 16), "abcd".hexAsEthereumAddressOrNull())
+        assertEquals(BigInteger("abcd", 16), "0xabcd".hexAsEthereumAddressOrNull())
+    }
+
+    @Test
+    fun hexAsEthereumAddress() {
+        assertThrow({ "ffffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddress() }, throwablePredicate = { it is InvalidAddressException })
+        assertThrow({ "0xffffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddress() }, throwablePredicate = { it is InvalidAddressException })
+        assertThrow({ "thisisporbablynotahexnumber".hexAsEthereumAddress() }, throwablePredicate = { it is InvalidAddressException })
+        assertEquals(BigInteger("ffffffffffffffffffffffffffffffffffffffff", 16), "ffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddress())
+        assertEquals(BigInteger("ffffffffffffffffffffffffffffffffffffffff", 16), "0xffffffffffffffffffffffffffffffffffffffff".hexAsEthereumAddress())
+        assertEquals(BigInteger("abcd", 16), "abcd".hexAsEthereumAddress())
+        assertEquals(BigInteger("abcd", 16), "0xabcd".hexAsEthereumAddress())
+    }
+
+    @Test
+    fun isValidEthereumAddressString() {
+        assertFalse("ffffffffffffffffffffffffffffffffffffffffff".isValidEthereumAddress())
+        assertFalse("0xffffffffffffffffffffffffffffffffffffffffff".isValidEthereumAddress())
+        assertFalse("thisisporbablynotahexnumber".isValidEthereumAddress())
+        assertFalse("abcd".isValidEthereumAddress())
```

Comment:
Are we sure we want this?

Currently the user always has to type all 40 chars. 

I am fine either way, but I think in some places we don't enforce this. Maybe this is a UX question.